File: src/CandidateTests.FeatureFlags.Api/Data/Flag.cs
namespace CandidateTests.FeatureFlags.Api.Data;

public class Flag
{
    public int Id { get; set; }

    public string Key { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public bool Enabled { get; set; }

    public int RolloutPercent { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }
}

File: web/types/flag.ts
export interface Flag {
  id: number
  key: string
  name: string
  enabled: boolean
  rolloutPercent: number
  createdAt: string
  updatedAt: string
}

export interface FlagInput {
  key: string
  name: string
  enabled: boolean
  rolloutPercent: number
}

export interface EvaluateResponse {
  enabled: boolean
}
